tests/Functional/Schema/MySQLSchemaManagerTest.php

```diff
@@ -231,23 +230,21 @@ public function testColumnCharsetChange(): void
             ->setLength(100)
             ->setNotnull(true)
             ->setPlatformOption('charset', 'utf8');
+        $this->dropAndCreateTable($table);
 
         $diffTable = clone $table;
         $diffTable->getColumn('col_string')->setPlatformOption('charset', 'ascii');
 
-        $fromSchema = new Schema([$table]);
-        $toSchema   = new Schema([$diffTable]);
-
         $diff = $this->schemaManager->createComparator()
-            ->compareSchemas($fromSchema, $toSchema)
-            ->toSql(
-                $this->connection->getDatabasePlatform()
-            );
+            ->diffTable($table, $diffTable);
+        self::assertNotNull($diff);
+        $this->schemaManager->alterTable($diff);
 
-        self::assertContains(
-            'ALTER TABLE test_column_charset_change CHANGE col_string'
-                . ' col_string VARCHAR(100) CHARACTER SET ascii NOT NULL',
```

Comment:
This patch affects the exact DDL used to alter the table: since now the column definition being compared always has an explicit charset and collation, it leaks into the resulting DDL (`COLLATION ascii_general_ci` is appended).

This is not a bug, since this is the default collation for the `ascii` charset according to the information schema.

Keeping this DDL unchanged would require additional juggling of column options. At the same time, a functional test shouldn't make assertions against SQL.